### ISSUE TITLE
flicker if plugin sets 'background' during startup

### ISSUE THREAD

**Issue by vimpostor** [neovim/neovim#19362]
### Neovim version (nvim -v)

v0.7.2

### Vim (not Nvim) behaves the same?

no, vim 9.0.0

### Operating system/version

Arch Linux

### Terminal name/version

wezterm nightly

### $TERM environment variable

xterm-256color

### Installation

Arch Linux community repo

### How to reproduce the issue

- Have a terminal emulator with light colorscheme
- Source a plugin that sets `background=light` as soon as the `plugin/pluginname.vim` is sourced
- Once nvim is started, the screen flickers briefly because neovim assumes `background=dark` and renders the UI before the plugin is sourced.

If we set `background=light` in `init.vim`, then we set it early enough before nvim renders anything. But it's not possible to do that from a plugin (without calling the plugin from `init.vim`), because the plugin is sourced too late in the startup process: nvim already shows the UI before sourcing it, while normal vim does not (so the problem does not occur with normal vim).

See the attached video for a visual explanation:
- neovim is on the left: Flickers visibly
- vim is on the right: Does not flicker

https://user-images.githubusercontent.com/21310755/178955503-66796c13-79ac-4e4b-8399-fd17b177194b.mp4

Minimal example `myplugin/plugin/myplugin.vim` (`myplugin` needs to be in `rtp`):
```vim
" myplugin.vim
set background=light
```

### Expected behavior

neovim renders with `background=light` immediately, like vim does

### Actual behavior

neovim renders with `background=dark`, then sources the plugin, rerenders with `background=light` causing a visible flickering

**Comment by vimpostor** [comment]
Unfortunately it looks like there is no way to execute something earlier than from `plugin` directories, is there?

```
	List of directories to be searched for these runtime files:
	  filetype.vim	filetypes by file name |new-filetype|
	  scripts.vim	filetypes by file contents |new-filetype-scripts|
	  autoload/	automatically loaded scripts |autoload-functions|
	  colors/	color scheme files |:colorscheme|
	  compiler/	compiler files |:compiler|
	  doc/		documentation |write-local-help|
	  ftplugin/	filetype plugins |write-filetype-plugin|
	  indent/	indent scripts |indent-expression|
	  keymap/	key mapping files |mbyte-keymap|
	  lang/		menu translations |:menutrans|
	  menu.vim	GUI menus |menu.vim|
	  pack/		packages |:packadd|
	  plugin/	plugin scripts |write-plugin|
	  print/	files for printing |postscript-print-encoding|
	  rplugin/	|remote-plugin| scripts
	  spell/	spell checking files |spell|
	  syntax/	syntax files |mysyntaxfile|
	  tutor/	tutorial files |:Tutor|
```

**Comment by justinmk** [comment]
> Unfortunately it looks like there is no way to execute something earlier than from `plugin/` directories, is there?

Only with `--cmd`. Can you try that , just for troubleshooting ? 

> neovim renders with `background=light` immediately, like vim does

Vim assumes bg=light on many environments. Does the opposite problem happen in Vim, with a dark terminal?

**Comment by zeertzjq** [comment]
I cannot reproduce this v0.9.5 or v0.10.0-dev. Does this bug still exist?

**Comment by vimpostor** [comment]
Yes I can still reproduce this with `v0.9.5` following the steps from the description (using a plugin that sets background from the `plugin/` directory).

**Comment by ErrorNoInternet** [comment]
I'm still reproducing this on v0.10.0-dev-78b000c, with the nord.nvim colorscheme. My current "fix" is to run `echo " "` right after setting the colorscheme

**Comment by pappasam** [comment]
I'm using:

```
NVIM v0.10.0-dev-2445+g2e1f5055a
Build type: RelWithDebInfo
LuaJIT 2.1.1707061634
Run "nvim -V1 -v" for more info
```

Based on the flicker I'm observing, this phenomenon appears to be consistent with this old issue: https://github.com/vim/vim/issues/11017

I confirm that adding `echo ' '` anywhere in `init.vim` resolves the issue. Very strange.

**Comment by frixaco** [comment]
I can confirm issue still exists on latest Neovim lightly.
 
Thank you, @ErrorNoInternet for the "fix"!

**Comment by ThorstenRhau** [comment]
This still happens on 0.10.3

**Comment by ManojPanda3** [comment]
in previous pr i think this get fixed @zeertzjq   sorry to disturb you  is this safe to close ? 
<video src="https://github.com/user-attachments/assets/283c2458-c4ec-4e01-b67b-8532f980a63d"></video>

**Comment by vimpostor** [comment]
I can confirm (via bisect) that this has been resolved with #28676 (note that the specific issue from above was also only reproducible inside tmux for me).
